onnxruntime/core/framework/utils.cc

```diff
@@ -23,9 +23,18 @@ AllocatorPtr GetAllocator(const SessionState& session_state, const OrtAllocatorI
   return session_state.GetExecutionProviders().GetAllocator(allocator_info);
 }
 
-common::Status AllocateHelper(const IExecutionProvider& execution_provider, int device_id, const Tensor& fetched_tensor,
+bool ProviderIsCpuBased(const std::string& provider_type) {
+  return provider_type == onnxruntime::kCpuExecutionProvider ||
+         provider_type == onnxruntime::kMklDnnExecutionProvider ||
```

Comment:
Can we avoid listing all the providers in the framework? Otherwise, it means people can't add a new CPU provider without changing the upstream onnxruntime source code.

Reply:
Also lets check all the places where enumeration or switch takes place and minimize the number of those.

---
In reply to: [310326829](https://github.com/microsoft/onnxruntime/pull/1553#discussion_r310326829) [](ancestors = 310326829)

Reply:
it will be avoided later, when we have a way to specify device from an execution provider.